homework/unique_ptr/Tests/tests.cpp

```diff
@@ -10,33 +10,46 @@ class TestObject {
     MOCK_METHOD(void, Destructor, (), (noexcept));
 };
 
-TEST(unique_ptr, destructor_test) {
+TEST(unique_ptr, default_constructor) {
+    auto smartPtr = UniquePointer<TestObject>();
+
+    EXPECT_EQ(nullptr, smartPtr.get());
+}
```

Comment:
This test is actually the same as the test for get(). Maybe there is a better way to distinguish the two.

Reply:
Did a quick check how to test the "const" case and it seems to be quite tricky. There is no straightforward way to test a program which doesn't compile.